src/grpc/subscription_operations.py

```diff
@@ -0,0 +1,115 @@
+import re
+import sqlalchemy as sa
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.ext.asyncio import AsyncSession
+from src.db.model import Subscription
+from src.grpc.generated.subscription_pb2 import (
+    CreateSubscriptionResponse,
+    GetSubscriptionsResponse,
+    ChangeSubscriptionResponse,
+    DeleteSubscriptionResponse,
+    ActivateSubscriptionResponse,
+    DeactivateSubscriptionResponse,
+)
+
+def validate_email(email: str) -> bool:
+    email_regex = r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)"
+    return re.match(email_regex, email) is not None
+
+async def create_subscription(session: AsyncSession, email: str, subscription_type: str):
+    try:
+        subscription = (
+            await session.execute(sa.select(Subscription).filter_by(email=email))
+        ).scalars().first()
+
+        if subscription is not None:
+            return CreateSubscriptionResponse(message="\nSubscription with this email already exists.")
+
+        if not validate_email(email):
+            return CreateSubscriptionResponse(message="\nInvalid email format")
+
+        session.add(Subscription(email=email, subscription_type=subscription_type))
+        await session.commit()
```

Comment:
session.commit() should be try excepted and a rollback issued of something goes wrong.

Reply:
This is not a good idea i would be worried that i have no idea when this exception might occur and if it would have predictable beheiviour.